src/tests/ftest/cart/dual_iface_server.c

```diff
@@ -0,0 +1,434 @@
+		other_rank = 1;
+	else
+		other_rank = 0;
+
+	rc = crt_group_primary_rank_add(crt_ctx[0], grp, other_rank,
+					other_server_uri);
+	if (rc != 0) {
+		D_ERROR("Failed to add rank=%d uri='%s'\n",
+			other_rank, other_server_uri);
+		assert(0);
+	}
+
+	server_ep.ep_rank = other_rank;
+	server_ep.ep_tag = tag;
+	server_ep.ep_grp = grp;
+
+	rc = crt_req_create(crt_ctx[0], &server_ep, RPC_PING, &rpc);
+	if (rc != 0) {
+		D_ERROR("crt_req_create() failed; rc=%d\n", rc);
+		assert(0);
+	}
+
+	input = crt_req_get(rpc);
+	input->src_rank = my_rank;
+	input->dst_tag = tag;
+
+	rc = crt_req_send(rpc, rpc_handle_reply, &sem);
+	if (rc != 0) {
+		D_ERROR("Failed to send rpc; rc=%d\n", rc);
+		assert(0);
+	}
+
+	tc_sem_timedwait(&sem, 10, __LINE__);
+	DBG_PRINT("Ping successful to rank=%d tag=%d\n", other_rank, tag);
+
+	/* Shutdown */
+	server_ep.ep_rank = other_rank;
+	server_ep.ep_tag = tag;
+	server_ep.ep_grp = grp;
+
+	rc = crt_req_create(crt_ctx[0], &server_ep, RPC_SHUTDOWN, &rpc);
+	if (rc != 0) {
+		D_ERROR("crt_req_create() failed; rc=%d\n", rc);
+		assert(0);
+	}
+
+	input = crt_req_get(rpc);
+	input->src_rank = my_rank;
+	input->dst_tag = tag;
+
+	rc = crt_req_send(rpc, rpc_handle_reply, &sem);
+	if (rc != 0) {
+		D_ERROR("Failed to send rpc; rc=%d\n", rc);
+		assert(0);
+	}
+
+	tc_sem_timedwait(&sem, 10, __LINE__);
+
+	/* Wait until shutdown is issued and progress threads exit */
+	for (i = 0; i < NUM_SERVER_CTX; i++)
+		pthread_join(progress_thread[i], NULL);
+
+	rc = crt_finalize();
+	if (rc != 0) {
+		D_ERROR("crt_finalize() failed with rc=%d\n", rc);
+		assert(0);
+	}
+
+	d_log_fini();
+
+	return rc;
+}
+
+static void
+print_usage(const char *msg)
+{
+	printf("Error: %s\n", msg);
+	printf("Usage: ./dual_iface_server [-i 'iface0,iface1' ");
+	printf("-d 'domain0,domain1' -p 'provider'\n");
+}
+
+int main(int argc, char **argv)
+{
+	int	pid;
+	int	fd0, fd1;
+	char	tmp_file0[] = "/tmp/server0-XXXXXX";
+	char	tmp_file1[] = "/tmp/server1-XXXXXX";
+	int	c;
+	char	*provider;
+	char	*arg_interface = NULL;
+	char	*arg_domain = NULL;
+	char	*arg_provider = NULL;
+	char	*iface0, *iface1;
+	char	*save_ptr = NULL;
+	char	*domain0, *domain1;
+	char	default_iface0[] = "ib0";
+	char	default_iface1[] = "ib1";
+	char	default_domain0[] = "mlx5_0";
+	char	default_domain1[] = "mlx5_1";
+	char	default_provider[] = "ofi+verbs;ofi_rxm\0";
+
+	while ((c = getopt(argc, argv, "i:p:d")) != -1) {
+		switch (c) {
+		case 'i':
+			arg_interface = optarg;
+			break;
+		case 'd':
+			arg_domain = optarg;
+			break;
+		case 'p':
+			arg_provider = optarg;
+			break;
+		default:
+			print_usage("");
+			return -1;
+		}
+	}
+
+	iface0 = default_iface0;
+	iface1 = default_iface1;
+	domain0 = default_domain0;
+	domain1 = default_domain1;
+	provider = default_provider;
+
+	if (arg_interface) {
+		iface0 = strtok_r(arg_interface, ",", &save_ptr);
+		if (iface0 == NULL) {
+			print_usage("Failed to parse iface0");
+			return -1;
+		}
+
+		iface1 = strtok_r(NULL, ",", &save_ptr);
```

Comment:
The invocation is "-p iface0,iface1".  So why are you looking for a second ","?

Reply:
good point, it is leftover from where params was single with all arguments coma separated; will change